### PR TITLE
Implement reporting to Distribution beacon for proper download counts

### PR DESCRIPTION
**Pull request by roland-chernov-akvelon**
This PR implements reporting of these parameters to Distribution beacon:
- Release id and distribution id for private groups.
- Release id and install id for public groups.

Similar PR for Android SDK: Microsoft/AppCenter-SDK-Android#614.

### CODE REVIEW

File: AppCenterDistribute/AppCenterDistribute/MSDistribute.m [hunk 1]
```diff
@@ -807,6 +876,9 @@ - (void)startDownload:(nullable MSReleaseDetails *)details {
           break;
         }
 
+        // Add downloaded release hash to report after installation.
```
Comment (TroubleMakerBen) [inline AppCenterDistribute/AppCenterDistribute/MSDistribute.m]: This comment is a little misleading as it's not guaranteed that app actually gets updated.

File: AppCenterDistribute/AppCenterDistribute/MSDistributePrivate.h [hunk 2]
```diff
@@ -84,6 +86,16 @@ static NSString *const kMSDistributionGroupIdKey = @"MSDistributionGroupId";
  */
 static NSString *const kMSUpdateSetupFailedPackageHashKey = @"MSUpdateSetupFailedPackageHash";
 
+/**
+ * The storage key for latest downloaded release hash.
+ */
+static NSString *const kMSDownloadedReleaseHashKey = @"MSDownloadedReleaseHash";
+
+/**
+ * The storage key for latest downloaded release The storage key for latest downloaded release ID.ID.
```
Comment (TroubleMakerBen) [inline AppCenterDistribute/AppCenterDistribute/MSDistributePrivate.h]: There's one ID too much ^^

File: AppCenterDistribute/AppCenterDistribute/MSDistribute.m [hunk 3]
```diff
@@ -271,11 +271,18 @@ - (void)checkLatestRelease:(NSString *)updateToken
 
     // Check if sender is still waiting for a response of the previous request.
```
Comment (TroubleMakerBen) [inline AppCenterDistribute/AppCenterDistribute/MSDistribute.m]: I don't think we need this comment anymore. What do you think?

File: AppCenterDistribute/AppCenterDistribute/MSDistributePrivate.h (hunk 2 above)
Comment (roland-chernov-akvelon) [reply AppCenterDistribute/AppCenterDistribute/MSDistributePrivate.h]: Oops. Fixed, thank you.

File: AppCenterDistribute/AppCenterDistribute/MSDistribute.m (hunk 1 above)
Comment (roland-chernov-akvelon) [reply AppCenterDistribute/AppCenterDistribute/MSDistribute.m]: Changed to:

> Store details to report new download after restart if this release is installed.

Please let me now if it's okay now.

File: AppCenterDistribute/AppCenterDistribute/MSDistribute.m (hunk 3 above)
Comment (roland-chernov-akvelon) [reply AppCenterDistribute/AppCenterDistribute/MSDistribute.m]: Removed.

**Comment by roland-chernov-akvelon** [comment]
@TroubleMakerBen I completed fixes that you mentioned. Could you please take a look.

**Comment by TroubleMakerBen** [comment]
@roland-chernov-akvelon looking good now! Can you resolve the merge conflict, please?

**Comment by roland-chernov-akvelon** [comment]
@TroubleMakerBen Sure, resolved.

File: AppCenterDistribute/AppCenterDistribute/MSDistributePrivate.h [hunk 4]
```diff
@@ -85,6 +87,16 @@ static NSString *const kMSDistributionGroupIdKey = @"MSDistributionGroupId";
  */
 static NSString *const kMSUpdateSetupFailedPackageHashKey = @"MSUpdateSetupFailedPackageHash";
 
+/**
+ * The storage key for latest downloaded release hash.
+ */
+static NSString *const kMSDownloadedReleaseHashKey = @"MSDownloadedReleaseHash";
+
+/**
+ * The storage key for latest downloaded release The storage key for latest downloaded release ID.
```
Comment (jaelim-ms) [inline AppCenterDistribute/AppCenterDistribute/MSDistributePrivate.h]: The sentence looks weird.

**Pull request merged**